content/updates/steps-to-break-up-a-terralith.md

```diff
@@ -0,0 +1,361 @@
+
+This command saves your state to a timestamped file. Remember to store your backups in a secure location!
+
+### #3 Create New Module Structures
+
+- For each new root module that you plan to introduce, create a separate directory.
+- Move the relevant resource configurations from the Terralith into the new root modules.
+- Initialize each new root module to ensure the backend is configured correctly.
+
+### #4 Inter-Module Communication (if any)
+
+If any root module depends on outputs from another, update the references using data sources, variables, or remote state. We recommend using data sources over [`terraform_remote_state`](https://developer.hashicorp.com/terraform/language/state/remote-state-data) whenever possible.
+
+For example, instead of retrieving subnets from a “network” module via [terraform_remote_state](https://registry.terraform.io/providers/hashicorp/terraform/latest/docs/data-sources/remote_state), you can use the AWS data source for subnets when provisioning a “database” module. This approach reduces coupling, keeps modules more autonomous, and ensures that changes in one module (like adding a subnet) don’t break another. Plus, any orchestration tool can trigger each root module independently, letting them fetch the latest infrastructure details directly from AWS without inter-module dependency on state files.
+
+### #5 Move Resources to The New State
+
+There are several ways to migrate the resources to their new state files depending on your TF version, the providers in use, and the data you must supply for imports.
+
+- **Option 1: Declarative Migrations Using TF 1.5+**
+
+  This is our preferred method. The idea here is to use declarative TF blocks to import resources into the new root modules state and then declaratively remove them from the old state.
+
+  **Importing existing resources**
+
+  In newer versions of TF, you can define [import](https://developer.hashicorp.com/terraform/language/import) blocks to move resources between modules in a declarative fashion. This method preserves resource history in Git and is more transparent and less error-prone than the alternative of manual state manipulation. This block is idempotent - once a resource is successfully imported, subsequent runs won’t trigger the same import action as long as that resource remains in that particular state file.
+
+  You can place `import` blocks in a separate file `imports.tf` in your new root module and then remove it after completing the migration.
+
+  While the import functionality is great, it gets even better. With Terraform or OpenTofu v1.7 or later, you can import several resources by iterating over a map or a set. Below is a sample configuration demonstrating how to use an import block with `for_each` for multiple GitHub repositories.
+
+  ```hcl
+  locals {
+    # Define a list of repository names to import
+    repos = {
+      "repo-foo": { description = “Our repo foo” },
+      "repo-bar": { description = “Our repo bar” },
```

Comment:
```suggestion
      "repo-foo": { description = "Our repo foo" },
      "repo-bar": { description = "Our repo bar" },
```

Reply:
Doh, I forgot that we've discussed this. Thanks for the fixing!